depthai_demo.py

```diff
@@ -360,6 +367,19 @@ def stop(self, *args, **kwargs):
             self._logOut.close()
         self.onTeardown(self)
 
+    def canRun(self):
+        return hasattr(self, "_device") and not self._device.isClosed()
+
+    def _logMonitorCallback(self, msg: dai.LogMessage):
+        if msg.level == dai.LogLevel.CRITICAL:
+            print(f"[CRITICAL] [{msg.time.get()}] {msg.payload}", file=sys.stderr)
+            sys.stderr.flush()
+            temperature = self._device.getChipTemperature()
+            if temperature.average > 105:
```

Comment:
Check if any temperatures is higher than 105C, otherwise it might trigger too late